### PR TITLE
[Service Bus] Connection Idle Timeout

### PR DESCRIPTION
**Pull request by jsquire**
# Summary

The focus of these changes is to add an option to the ServiceBusClient to customize the idle timeout.  Previously, this was a hard-coded value of 60 seconds; that value has been preserved as the default.

# References and Related

- [Allow setting idle timeout in ServiceBus client (#35663)](https://github.com/Azure/azure-sdk-for-net/issues/35663)
- [Event Hubs - EventHubConnectionOptions.ConnectionIdleTimeout](https://learn.microsoft.com/dotnet/api/azure.messaging.eventhubs.eventhubconnectionoptions.connectionidletimeout?view=azure-dotnet#azure-messaging-eventhubs-eventhubconnectionoptions-connectionidletimeout)

### CODE REVIEW

**Review by JoshLove-msft — APPROVED** [review]
Is it worth raising this in the office hours in case there is some difference in the service behavior between SB and EH?

**Comment by jsquire** [comment]
> Is it worth raising this in the office hours in case there is some difference in the service behavior between SB and EH?

Not sure that I'm following.   What behavior would be different?

**Comment by JoshLove-msft** [comment]
> > Is it worth raising this in the office hours in case there is some difference in the service behavior between SB and EH?
> 
> Not sure that I'm following. What behavior would be different?

I know that there is service-specific behavior when it comes to the link-level idle timeout. Could there be something similar for connections?

**Review by m-redding — APPROVED** [review]

File: sdk/servicebus/Azure.Messaging.ServiceBus/src/Client/ServiceBusClientOptions.cs [hunk 1]
```diff
@@ -56,6 +57,34 @@ public class ServiceBusClientOptions
...
 
+        /// <summary>
+        ///   The amount of time to allow a connection to have no observed traffic before considering
+        ///   it idle and eligible to close.
+        /// </summary>
+        ///
+        /// <value>The default idle timeout is 60 seconds.  The timeout must be a positive value.</value>
+        ///
+        /// <remarks>
+        ///   If a connection is closed due to being idle, the Event Hubs clients will automatically
```
Comment (lynn-orrell) [inline sdk/servicebus/Azure.Messaging.ServiceBus/src/Client/ServiceBusClientOptions.cs:68]: I believe this needs to be changed to "Service Bus clients"...

Comment (lynn-orrell) [reply sdk/servicebus/Azure.Messaging.ServiceBus/src/Client/ServiceBusClientOptions.cs:68]: I'm just impressed at how quickly you got this fix in. Thanks again!

Comment (jsquire) [reply sdk/servicebus/Azure.Messaging.ServiceBus/src/Client/ServiceBusClientOptions.cs:68]: Thanks for the heads-up.   ##35958 should fix that.

**Pull request merged**